core/src/Borrower.sol

```diff
@@ -82,14 +74,14 @@ contract Borrower is IUniswapV3MintCallback {
         packedSlot.owner = owner;
     }
 
-    function liquidate(ILiquidator callee, bytes calldata data) external {
+    function liquidate(ILiquidator callee, bytes calldata data, uint256 strain) external {
```

Comment:
what is the purpose of `strain`?

Reply:
Good question (and I will write docstrings to answer it too) -- previously, the liquidation callee decided what percentage of remaining liabilities to obtain through swapping, and informed `Borrower` through the callback return values (`converted0` and `converted1`). `Borrower` then computed a reward. Now, since I've switched the order, `Borrower` needs to know ahead of time how much the callee intends to swap.

When `strain == 1`, the callee is expected to swap enough assets to cover 100% of debt. When `strain == 2` it's 50%, `strain == 3` is 33%, and so on. In almost every case, liquidators will use `strain = 1` because that maximizes their reward. But if for some reason the market is "strained" or the debt is huge, and a swap that large can't go through, they can opt to only cover a portion of the debt by setting higher strain values.